lib/db.js

```diff
@@ -1,44 +1,54 @@
-var logger = require('juttle/lib/logger').getLogger('sql-db-init');
-var _ = require('underscore');
-var knex;
-var savedDbId;
-var confArr;
-
-var DB = {
-    init: function(config) {
+'use strict';
+
+/* global JuttleAdapterAPI */
+let logger = JuttleAdapterAPI.getLogger('sql-db-init');
+
+let _ = require('underscore');
+
+let knex;
+let savedDbId;
+let confArr;
+
+class DB {
+    static init(config) {
         confArr = _.isArray(config) ? config : [config];
-    },
+    }
 
-    getDbConnection: function (options) {
+    static getDbConnection(options) {
         options = options || {};
 
-        var dbid = `${options.id}-${options.db}`;
+        let dbid = `${options.id}-${options.db}`;
         if (knex && dbid === savedDbId) {
             return knex;
         }
 
-        var conf = _.findWhere(confArr, {id: options.id}) || confArr[0];
-        knex = DB.getKnex(conf, options);
+        let conf = _.findWhere(confArr, {id: options.id}) || confArr[0];
+
+        //ensure that no changes are permanently made to the config in getKnex function.
+        conf = _.clone(conf);
+
+        knex = this.getKnex(conf, options);
         savedDbId = dbid;
 
         if (knex.client) {
             logger.info('initializing db connection with conf:', knex.client.config);
         } else {
+            knex = null;
```

Comment:
Yes but it is a static method. If you call the function again you could find that `knex` is still set and the error will not be thrown.